web/react/components/sidebar.jsx

```diff
@@ -322,7 +304,35 @@ export default class Sidebar extends React.Component {
             showBottomUnread
         });
     }
-    createChannelElement(channel, index) {
+
+    handleLeaveDirectChannel(channel) {
+        if (!this.isLeaving.get(channel.id)) {
+            this.isLeaving.set(channel.id, true);
+
+            const preference = PreferenceStore.setPreference(Constants.Preferences.CATEGORY_DIRECT_CHANNEL_SHOW, channel.teammate_id, 'false');
+            AsyncClient.savePreferences(
```

Comment:
Just change this to Client.savePreferences and remove the success and error parameters from AsyncClient